search/query.go

```diff
@@ -114,7 +122,51 @@ func ParseQuery(q string) Query {
 	return Query{Term: strings.Join(terms, " "), ExactTerms: exactTerms, Original: q, Filters: filters}
 }
 
-func createResultsQuery(resultTypes []string, q Query, docIds []string, filterOutCUSources []string) elastic.Query {
+func createSpanNearQuery(field string, term string, boost float32, slop int, inOrder bool) (elastic.Query, error) {
+	clauses := make([]string, 0)
+	spanNearMask := `{"span_near": { "clauses": [%s], "slop": %d, "boost": %f, "in_order": %t } }`
+	clauseMask := `{"span_multi": { "match": { "fuzzy": { "%s": { "value": "%s", "fuzziness": %s, "transpositions": %s } } } } }`
+	splitted := strings.Fields(term)
+	for _, t := range splitted {
+		if t == "<" || t == ">" || t == "-" {
+			continue
+		}
+		fuzzines := `"AUTO"`     // Default.
+		transpositions := "true" // Default.
+		runes := []rune(t)
+		_, convertToIntErr := strconv.Atoi(t)
+		if convertToIntErr == nil || (len(runes) == 3 && runes[1] == '"') || (len(runes) == 4 && runes[2] == '"') {
+			//  We dont use fuzzines for numeric values (number or hebrew numeric representation like מ"ג)
+			fuzzines = "0"
+		} else if len(runes) == 1 && runes[0] >= 'א' && runes[0] <= 'ת' {
+			// This logic allows finding single hebrew letter with ' symbol without the mention of the ' symbol.
+			// The solution is not perfect for all times. In some (rare) cases the letter may be replaced with another latter: ג' קווים - ד
+			fuzzines = "1"
+			transpositions = "false" // Limit the fuzzines not to include transpositions of two adjacent characters (ח' -> 'ח). Maybe not required.
+		}
+		b, err := json.Marshal(t)
+		if err != nil {
+			return nil, errors.Wrap(err, "createSpanNearQuery")
+		}
+		// Trim the beginning and trailing " character
+		esc := string(b[1 : len(b)-1])
+		clause := fmt.Sprintf(clauseMask, field, esc, fuzzines, transpositions)
+		clauses = append(clauses, clause)
+	}
+	clausesStr := strings.Join(clauses, ",")
+	queryStr := fmt.Sprintf(spanNearMask, clausesStr, slop, boost, inOrder)
+	//fmt.Printf("SpanNear Query: %s\n", queryStr)
+	query := elastic.NewRawStringQuery(queryStr)
+	return query, nil
+}
+
+// Creates a result query for elastic.
+// resultTypes - list of search result types: sources, topics, CU's, etc..
+// docIds - optional list of _uid's for filtering the search. If the parameter value is nil, no filtering is applied.
+// filterOutCUSources - optional list of sources for which we want to filter out the CU's that connected to those sources
+//	(in order to avoid duplication between carousel and regular results).
+// titlesOnly - limit our search only to title fields: title, full_title and description in case we search for intent sources. Used for intent search.
+func createResultsQuery(resultTypes []string, q Query, docIds []string, filterOutCUSources []string, titlesOnly bool) (elastic.Query, error) {
```

Comment:
You explained only the input params, not the query structure. Please add comment with query structure:

Query Structure:
- Filter by result types (programs/lessons/sources...).
- Filter by doc ids (specific list of ids for second highlight run).
- Filter by cu sources  ........ how to explain?

Term - Match query on
  Title, Description, Content
  Will score MAX between: SLOP, Boost on exact match, and on title .... Few words about createSpanNearQuery
ExactTerm - Exact matching "this should exactly match"
 ....

Reply:
Few words about decay:
AddScoreFunc(elastic.NewGaussDecayFunction().FieldName("effective_date").Decay(0.6).Scale("2000d")), nil

And min score ... why?